src/seqtk/seqtk_subseq/script.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+## VIASH START
+## VIASH END
+
+# Function to check if a file is a valid BED file
+check_bed_file() {
+    local file="$1"
+    num_columns=$(head -n 1 "$file" | cut -f 1- | tr '\t' '\n' | wc -l)
+
+    # Check if the file exists
+    if [[ ! -f "$file" ]]; then
+        echo "Error: The specified file does not exist."
+        return 1
+    fi
+
+    # Check if the file is non-empty
+    if [[ ! -s "$file" ]]; then
+        echo "Error: The specified file is empty."
+        return 1
+    fi
+
+    # Check if the file is a valid BED file (minimum 3 tab-separated columns)
+    if [[ $num_columns -lt 3 ]]; then
+        echo "The specified file is not a valid BED file. It should have at least three tab-separated columns."
+        return 1
+    fi
+
+    # Additional check: Ensure that the 6th column (if present) is either + or -
+    if [[ $num_columns -eq 6 ]]; then
+        while IFS=$'\t' read -r -a columns; do
+            columns[5]=${columns[5]%$'\n'}
+            if [[ ${columns[5]} == "+" || ${columns[5]} == "-" ]]; then
+                return 0
+            else 
+                echo "Error: The 6th column of the specified BED file should be either + or -."
+                echo "Offending line: ${columns[5]}"
+                return 1
+            fi
+        done < "$file"
+    fi
+
+    return 0
+}
+
+# Function to check if a file is a valid list of FASTA file IDs
+check_fasta_id_list() {
+    local file="$1"
+
+    # Check if the file exists
+    if [[ ! -f "$file" ]]; then
+        echo "Error: The specified file does not exist."
+        return 1
+    fi
+
+    # Check if the file is non-empty
+    if [[ ! -s "$file" ]]; then
+        echo "Error: The specified file is empty."
+        return 1
+    fi
+
+    # Additional check: Ensure that each line contains only one word (FASTA ID)
+    if ! awk 'NF != 1 { exit 1 }' "$file"; then
+        return 1
+    fi
+
+    return 0
+}
+
+# Check if the par_name_list is given and validate accordingly
+if [[ -n "$par_name_list" ]]; then
+    if check_fasta_id_list "$par_name_list"; then
+        echo "The specified file is a valid list of FASTA IDs."
+    elif check_bed_file "$par_name_list"; then
+        echo "The specified file is a valid BED file."
+    else
+        echo "Error: The specified file is neither a valid BED file nor a valid list of FASTA IDs."
+        exit 1
+    fi
+fi
```

Comment:
Let me think about this for a bit. While I do appreciate the extra sanity checks, I'm worried that you might be introducing behaviour that is different from what the tool already provides. In addition, its something that requires more maintenance.

I'm inclined to not do any sanity check other than passing the arguments to the underlying tool, and assuming that the tool does the necessary sanity checks.

WDYT?

Reply:
I think you are right, It is extra work. I added because the tool does not really check format of the list passed as argument.

Reply:
Apologies for the extra work!

Having thought some more about it, I think it _might_ be a good idea to create a set of helper bash scripts, e.g.

* src/_helpers/file.sh
* src/_helpers/text.sh
* src/_helpers/fasta.sh
* src/_helpers/bed.sh

Which use standard linux tools to do some checks. These could be sourced in the unit tests to alleviate some of the work of checking for contents of the files, e.g.

```bash
# load helper functions
source "$meta_resources_dir/_helpers/file.sh"
source "$meta_resources_dir/_helpers/fasta.sh"

cat > test.fasta << HERE
some fasta
HERE

$meta_executable --input test.fasta --output out.fasta

assert_file_exists out.fasta
assert_file_not_empty out.fasta
assert_fasta_somecheck out.fasta "..."
```

In this way, we could provide a couple of basic assertions in a way that would still make it maintainable.

WDYT? Feel free to turn this into a separate issue :relaxed:

Reply:
This was my idea because `seqtk` doesn't throw any kind of error when the .bed file is not valid but simply returns an incorrect result